MitoMathHelper.js

```diff
@@ -4,82 +4,143 @@
+        let distance = this._distanceBetweenTwoPointsNoSqrt(closestPoint, circleB.getPosition());
         let radiusTotal = circleA.getRadius() + circleB.getRadius();
-        if(radiusTotal >= distance){
-            let backOff = Math.sqrt((radiusTotal*radiusTotal) - (distance*distance));
-            let backPoint = [backOff *( combineVelocity[0] / combineMagnitude) , backOff *( combineVelocity[1] / combineMagnitude)];
-            let closestPointWithinVelocity = this.addPoints(closestPoint , this.invertPoint(backPoint));
-            let distanceBetween = this.distanceBetweenTwoPoints(circleA.getPosition(), closestPointWithinVelocity);
+        radiusTotal *= radiusTotal;
+        if (radiusTotal >= distance) {
+            let backOff = Math.sqrt(radiusTotal - distance);
+            let backPoint = [backOff * (combineVelocity[0] / combineMagnitude), backOff * (combineVelocity[1] / combineMagnitude)];
+            let closestPointWithinVelocity = this._addPoints(closestPoint, this._invertPoint(backPoint));
+            let distanceBetween = this._distanceBetweenTwoPoints(circleA.getPosition(), closestPointWithinVelocity);
             let tickPercentage = distanceBetween / combineMagnitude;
-            console.assert(tickPercentage >= 0 && tickPercentage <= 1, "detectCollisionTime: value outside of range: " + tickPercentage );
+            if (tickPercentage < 0 || tickPercentage > 1) {
+                return null;
+            }
             return tickPercentage;
-        }else{
-            return null;
         }
+        return null;
+    }
 
+    /**
+     * returns distance ^ 2
+     * @param pointA
+     * @param pointB
+     * @returns {number}
+     * @private
+     */
+    static _distanceBetweenTwoPointsNoSqrt(pointA, pointB) {
+        let xDifference = pointB[0] - pointA[0];
+        let yDifference = pointB[1] - pointA[1];
+        return xDifference * xDifference + yDifference * yDifference;
     }
 
-    distanceBetweenTwoPoints(pointA, pointB){
+    /**
+     * returns distance between two points
+     * @param pointA
+     * @param pointB
+     * @returns {number}
+     * @private
+     */
+    static _distanceBetweenTwoPoints(pointA, pointB) {
         return Math.hypot(pointB[0] - pointA[0], pointB[1] - pointA[1]);
     }
 
-    detectCollisionCirCir(circleA, circleB){
+    /**
+     * detects if two circles overlap
+     * @param pointA
+     * @param radiusA
+     * @param pointB
+     * @param radiusB
+     * @returns {boolean}
+     * @private
+     */
+    static _detectCollision(pointA, radiusA, pointB, radiusB) {
         let distance = Math.hypot(
-            circleB.getPosition()[0] - circleA.getPosition()[0],
-            circleB.getPosition()[1] - circleA.getPosition()[1]);
-        let radiusTotal = circleA.getRadius() + circleB.getRadius();
-        return  radiusTotal >= distance;
+            pointA[0] - pointB[0],
+            pointA[1] - pointB[1]);
```

Comment:
pointA[1] - pointB[1],
);

you don't need to do sqrt here, find all places where you don't need to do sqrt